blog/content/post/2024-02-07-cross2024_ftpm_tee_ta.md

```diff
@@ -0,0 +1,112 @@
+---
+title: Embracing fTPM on embedded ARM Devices - Insights and Solutions
+abstract: 'This article examines the integration of firmware Trusted Platform Module (fTPM) in embedded devices, particularly for custom ARM-based projects. It covers the benefits alongside challenges and potential drawbacks. The piece compares fTPM with other TPM implementations, offering strategies to mitigate issues and guide developers towards informed choices and a stronger open security framework for embedded devices.'
+cover: /covers/image-file.png
+author: tymek.burak
+layout: post
+published: true
+date: 2024-02-07
+archives: "2024"
+
+tags:
+  - ftpm
+  - tpm
+  - tee
+  - op-tee
+  - arm
+  - trustzone
+  - confidential
+  - computing
+categories:
+  - IoT
+  - Security
+  - Manufacturing
+
+---
+
+In the realm of digital security, Trusted Platform Modules (TPMs) play a
+pivotal role by safeguarding cryptographic keys and other sensitive data used
+in authentication processes. Traditionally, TPMs are external hardware
+components, adding a layer of physical security but also increasing the overall
+cost of device manufacturing. However, in the quest to balance
+cost-effectiveness with security, one viable solution that emerges is the
+integration of firmware-based TPMs (fTPMs) within the CPU itself. This approach
+presents an innovative alternative for scenarios where traditional hardware
+TPMs may not be feasible or cost-effective but also come with fallbacks.
+Fundamentally it requires the establishment of a Trusted Execution Environment
```

Comment:
@tym2k1 is TEE always required to implement fTPM? How about fTPM on other architectures as x86? I am not sure is TEE is something that is required everytime.

Reply:
You're right its not. [addressed](https://github.com/3mdeb/news-and-ideas/pull/581/files#diff-e2d90bda5063566a842217b25fc0360447359f10ef41e81af571c35f2a749777R39-R42)